test/support/mock_fs_helper.ex

```diff
@@ -0,0 +1,49 @@
+defmodule MixTestWatch.MockFsHelper do
```

Comment:
I'm suggesting just testing with paths for files that exist or do not, and forgetting about creating Mick data at all. I realise now that since the top level directory is checked we can't really use a fixtures directory, but we can use real files just as happily.

Reusing data between tests is fine I think. The complexity added in mocking or dynamically creating files isn't worth the gain in separation and slight reduction in brittleness.

Yee, with real files we can run async.